src/transformers/models/gemma3/modeling_gemma3.py

```diff
@@ -1066,20 +1100,19 @@ def forward(self, vision_outputs: torch.Tensor):
 
 
 @add_start_docstrings(
-    """The GEMMA3 model which consists of a vision backbone and a language model.""",
+    """Base Gemma3 model which consists of a vision backbone and a language model withou language modeling head.""",
     GEMMA3_START_DOCSTRING,
 )
-class Gemma3ForConditionalGeneration(Gemma3PreTrainedModel, GenerationMixin):
+class Gemma3Model(Gemma3PreTrainedModel):
+    _key_mapping = {"language_model.model": "language_model"}
```

Comment:
as this appears in the model class, I would be happy to have a more meaningful name here! 
`_on_the_fly_conversion_mapping` or something like this!